prepatch mount fixes

slemke opened the pull request:
This PR fixes two open issues:

* #289: I moved the brutosaur mount to BMAH and upgraded it to a section instead of a subsection, because I also added mounts from these lists:
  * https://www.warcraftmounts.com/bmah.php
  * https://www.wowhead.com/news=318099/black-market-auction-house-location-in-shadowlands
* #285: I marked the old RAF mounts as unobtainable

I've done this by hand for now.

seirl reviewed (changes requested):
This breaks the assumption that each mount should only show up once in the list. I know it's sometimes weird, but I think we need to find the most relevant section for each mount, instead of putting each mount in all the sections that match.

slemke commented:
I see, the BMAH category is now removed, because all those mounts are already in some other category. So I only marked the old RAF mounts unobtainable and moved the brutosaur to BMAH in this PR.
